CustomTypes.cpp

```diff
@@ -4,83 +4,124 @@
  * terms of the license can be found in the LICENSE.txt file.
  */
-
 #include "CustomTypes.h"
 
-constexpr const char* AllTypesSource = R"(
-typedef void* tptr_t;
-typedef void* fptr_t;
-typedef int32_t rptr_t;
-
-typedef void* id;
-typedef char* SEL;
-
-typedef char BOOL;
-typedef int64_t NSInteger;
-typedef uint64_t NSUInteger;
-typedef double CGFloat;
-
-struct CFString {
-    const tptr_t isa;
-    uint64_t flags;
-    const tptr_t data;
-    uint64_t size;
-};
-
-struct objc_method_entry_t {
-    rptr_t name;
-    rptr_t types;
-    rptr_t imp;
-};
-
-struct objc_method_t {
-    tptr_t name;
-    tptr_t types;
-    tptr_t imp;
-};
-
-struct objc_method_list_t {
-    uint32_t obsolete;
-    uint32_t count;
-};
-
-struct objc_class_ro_t {
-    uint32_t flags;
-    uint32_t start;
-    uint32_t size;
-    uint32_t reserved;
-    const tptr_t ivar_layout;
-    const tptr_t name;
-    const tptr_t methods;
-    const tptr_t protocols;
-    const tptr_t vars;
-    const tptr_t weak_ivar_layout;
-    const tptr_t properties;
-};
-
-struct objc_class_t {
-    const tptr_t isa;
-    const tptr_t super;
-    void* cache;
-    void* vtable;
-    const fptr_t data;
-};
-)";
-
 namespace CustomTypes {
 
 using namespace BinaryNinja;
 
-void defineAll(Ref<BinaryView> bv)
+/**
+ * Helper method for defining structure types
+ */
+std::pair<QualifiedName, Ref<Type>> finalizeStructureBuilder(Ref<BinaryView> bv, StructureBuilder sb, std::string name)
 {
-    std::map<QualifiedName, Ref<Type>> types, variables, functions;
-    std::string errors;
+    auto classTypeStruct = sb.Finalize();
+
+    QualifiedName classTypeName(name);
+    std::string classTypeId = Type::GenerateAutoTypeId("objc", classTypeName);
+    Ref<Type> classType = Type::StructureType(classTypeStruct);
+    QualifiedName classQualName = bv->DefineType(classTypeId, classTypeName, classType);
 
-    bv->GetDefaultPlatform()->ParseTypesFromSource(AllTypesSource,
-        "ObjectiveNinja.h", types, variables, functions, errors);
+    return {classQualName, classType};
+}
 
-    for (const auto& [name, type] : types)
-        bv->DefineUserType(name, type);
+/**
+ * Helper method for defining typedefs
+ */
+inline void defineTypedef(Ref<BinaryView> bv, const QualifiedName name, Ref<Type> type)
+{
+    std::string typeID = Type::GenerateAutoTypeId("objc", name);
+    bv->DefineType(typeID, name, type);
 }
 
+void defineAll(Ref<BinaryView> bv)
+{
+    int addrSize = bv->GetAddressSize();
+
+    defineTypedef(bv, {"id"}, Type::PointerType(addrSize, Type::VoidType()));
+    defineTypedef(bv, {"SEL"}, Type::PointerType(addrSize, Type::IntegerType(1, false)));
+
+    defineTypedef(bv, {"BOOL"}, Type::IntegerType(1, false));
+    defineTypedef(bv, {"NSInteger"}, Type::IntegerType(addrSize, true));
+    defineTypedef(bv, {"NSUInteger"}, Type::IntegerType(addrSize, false));
+    defineTypedef(bv, {"CGFloat"}, Type::FloatType(addrSize));
+
+    StructureBuilder cfstringStructBuilder;
+
```

Comment:
Please remove the blank lines between `StructureBuilder xxxStructBuilder`, the calls to `AddMember`, and the ultimate `finalizeStructureBuilder` call. Doing so makes the source more quickly scannable